### PR TITLE
Enable in-cluster Eventing without creating Kyma Applications

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

Enable in-cluster Eventing without creating Kyma Applications:
- Update the Event publishers to support publishing in-cluster legacy events.
- Update the Eventing controllers (eventing-controller-nats and eventing-controller), to be able to create NATS Subjects and Webhooks (respectively) event if the given Kyma subscription contains an event-type for non-existing Kyma Application.

**Related issue(s)**
#10669

### CODE REVIEW

File: components/eventing-controller/pkg/handlers/eventtype/clean_test.go [hunk 1]
```diff
@@ -25,60 +26,65 @@ func TestCleaner(t *testing.T) {
...
+			givenApplicationName: "testapp",
+			givenEventType:       "sap.kyma.testapp.order.created", // missing the version
 			wantError:            true,
 		},
 		{
 			givenEventTypePrefix: "sap.kyma",
 			givenApplicationName: "testapp",
-			givenEventType:       "sap.kyma.test-app.order.created.v1",
-			wantEventType:        "sap.kyma.testapp.order.created.v1",
+			givenEventType:       "sap.testapp.order.created.v1", // prefix not matched
```
Comment (sayanh) [inline components/eventing-controller/pkg/handlers/eventtype/clean_test.go:79]: How about using `name` for the testCase struct which will explain the context?

File: components/eventing-controller/pkg/handlers/eventtype/clean_test.go [hunk 2]
```diff
@@ -25,60 +26,65 @@ func TestCleaner(t *testing.T) {
 			givenApplicationName: "testapp",
 			givenEventType:       "sap.kyma.testapp.order.created.v1",
 			wantEventType:        "sap.kyma.testapp.order.created.v1",
+			wantError:            false,
```
Comment (sayanh) [inline components/eventing-controller/pkg/handlers/eventtype/clean_test.go:29]: What about adding tests for Clean() with empty application list?

File: components/eventing-controller/pkg/handlers/eventtype/clean_test.go (hunk 1 above)
Comment (sayanh) [inline components/eventing-controller/pkg/handlers/eventtype/clean_test.go:79]: How about adding `name` for the testCase struct to provide some context and later use `t.Run("...context...", func(){})`?

File: components/eventing-controller/pkg/handlers/eventtype/clean_test.go (hunk 2 above)
Comment (marcobebway) [reply components/eventing-controller/pkg/handlers/eventtype/clean_test.go:31]: Done

File: components/eventing-controller/pkg/handlers/eventtype/clean_test.go (hunk 1 above)
Comment (marcobebway) [reply components/eventing-controller/pkg/handlers/eventtype/clean_test.go]: Done

**Review by sayanh — APPROVED** [review]

**Comment by pbochynski** [comment]
Is there any documentation how to use in-cluster messaging? What is the recommendation about event types? Can we publish directly to nats?

**Comment by marcobebway** [comment]
> Is there any documentation how to use in-cluster messaging? What is the recommendation about event types? Can we publish directly to nats?

This will be done in this PR: https://github.com/kyma-project/kyma/pull/10820.

**Review by sayanh — APPROVED** [review]
